frontend/src/server/api/routers/dataset.ts

```diff
@@ -13,12 +13,10 @@ import {
 import { z } from "zod";
 
 export const datasetRouter = createTRPCRouter({
-  search: protectedProcedure
+  search: publicProcedure
     .input(SearchDatasetSchema)
-    .query(async ({ input, ctx }) => {
-      const user = ctx.session.user;
-      const apiKey = user.apikey;
-      const searchResults = await searchDatasets({ apiKey, options: input });
+    .query(async ({ input }) => {
+       const searchResults = await searchDatasets({ options: input });
```

Comment:
If there is no apiKey, you don't send, but if there is, you send.

Reply:
fixed